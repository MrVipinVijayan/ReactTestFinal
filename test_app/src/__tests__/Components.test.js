import { cleanup } from '@testing-library/react';
import HomeRouterComponent from '../routes/HomeRouterComponent';
import DashboardComponent from '../pages/DashboardComponent';
import RegisterStudentComponent from '../pages/RegisterStudentComponent';
import StudentsListComponent from '../pages/ViewStudentListComponent';
import StudentComponent from '../components/StudentComponent';
import App from '../App';
import renderer from 'react-test-renderer';
import store from '../store';

import configureMockStore from 'redux-mock-store';

import { Provider } from 'react-redux';

const mockStore = configureMockStore();

afterEach(cleanup);

it('renders dashboard component correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<App>
					<DashboardComponent />
				</App>
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders home router component correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<App>
					<HomeRouterComponent />
				</App>
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders add student component correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<App>
					<RegisterStudentComponent />
				</App>
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders view students component correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<App>
					<StudentsListComponent />
				</App>
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders student component correctly', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<App>
					<StudentComponent />
				</App>
			</Provider>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
