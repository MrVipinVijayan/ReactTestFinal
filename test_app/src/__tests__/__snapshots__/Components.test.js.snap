// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add student component correctly 1`] = `
<div
  className="App"
>
  <nav>
    Student Registration
    <button
      onClick={[Function]}
    >
      X
    </button>
  </nav>
  <div>
    <a
      href="/add"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        Register Student
      </button>
    </a>
       
    <a
      href="/view"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        View Students
      </button>
    </a>
  </div>
</div>
`;

exports[`renders dashboard component correctly 1`] = `
<div
  className="App"
>
  <nav>
    Student Registration
    <button
      onClick={[Function]}
    >
      X
    </button>
  </nav>
  <div>
    <a
      href="/add"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        Register Student
      </button>
    </a>
       
    <a
      href="/view"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        View Students
      </button>
    </a>
  </div>
</div>
`;

exports[`renders home router component correctly 1`] = `
<div
  className="App"
>
  <nav>
    Student Registration
    <button
      onClick={[Function]}
    >
      X
    </button>
  </nav>
  <div>
    <a
      href="/add"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        Register Student
      </button>
    </a>
       
    <a
      href="/view"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        View Students
      </button>
    </a>
  </div>
</div>
`;

exports[`renders student component correctly 1`] = `
<div
  className="App"
>
  <nav>
    Student Registration
    <button
      onClick={[Function]}
    >
      X
    </button>
  </nav>
  <div>
    <a
      href="/add"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        Register Student
      </button>
    </a>
       
    <a
      href="/view"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        View Students
      </button>
    </a>
  </div>
</div>
`;

exports[`renders view students component correctly 1`] = `
<div
  className="App"
>
  <nav>
    Student Registration
    <button
      onClick={[Function]}
    >
      X
    </button>
  </nav>
  <div>
    <a
      href="/add"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        Register Student
      </button>
    </a>
       
    <a
      href="/view"
      onClick={[Function]}
    >
      <button
        className="sc-bdnxRM bjwUQJ"
      >
        View Students
      </button>
    </a>
  </div>
</div>
`;
